packages/slate/src/models/node.js

```diff
@@ -198,12 +198,18 @@ class Node {
     first = assertKey(first)
     second = assertKey(second)
 
-    const keys = this.getKeysAsArray()
-    const firstIndex = keys.indexOf(first)
-    const secondIndex = keys.indexOf(second)
-    if (firstIndex == -1 || secondIndex == -1) return null
-
-    return firstIndex < secondIndex
+    if (first === second) return false
+    if (parseInt(first, 10) > parseInt(second, 10)) {
+      // Ensure areDescendantSorted(second, first) is also cached
+      // Always prefer newer node in second argument, for potential
+      // futher optimization
+      return !this.areDescendantsSorted(second, first)
+    }
+    if (first === this.key) return true
+    if (second === this.key) return false
```

Comment:
Because they have different return; we can write 
```
  if (first === this.key || second === this.key) return first === this.key
```

But I am not sure whether it is easier to understand